src/code/game.c

```diff
@@ -0,0 +1,530 @@
+            HREG(83) = 0;
+        }
+        if (HREG(81) < 0) {
+            HREG(81) = 0;
+            // & 0xFFFFFFFF necessary for matching.
+            hexDumpSize = (HREG(83) == 0 ? 0x100 : HREG(83) * 0x10) & 0xFFFFFFFF;
+            LogUtils_LogHexDump(PHYSICAL_TO_VIRTUAL(HREG(82) << 8), hexDumpSize);
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+// Regalloc mostly
+void GameState_DrawInputDisplay(u16 input, Gfx** gfx) {
+    static const u16 sInpDispBtnColors[] = {
+        0xFFC1, 0xFFC1, 0xFFC1, 0xFFC1, 0x7BDF, 0x7BDF, 0x07FF, 0xF83F,
+        0x7BDF, 0x7BDF, 0x7BDF, 0x7BDF, 0xF801, 0x7BDF, 0x07C1, 0x003F,
+    };
+    s32 i, j;
+    s32 lrx, lry, ulx, uly;
+    Gfx* gfxP = *gfx;
+    gDPPipeSync(gfxP++);
+    gDPSetOtherMode(gfxP++,
+                    G_AD_PATTERN | G_CD_MAGICSQ | G_CK_NONE | G_TC_CONV | G_TF_POINT | G_TT_NONE | G_TL_TILE |
+                        G_TD_CLAMP | G_TP_NONE | G_CYC_FILL | G_PM_NPRIMITIVE,
+                    G_AC_NONE | G_ZS_PIXEL | G_RM_NOOP | G_RM_NOOP2);
+
+    for (i = 0, j = 1; i < 0x10; i++, j++) {
+        if (input & (1 << i)) {
+            gDPSetFillColor(gfxP++, (sInpDispBtnColors[i] << 0x10) | sInpDispBtnColors[i]);
+            ulx = i * 4 + 226;
+            uly = 220;
+            lrx = j * 4 + 225;
+            lry = 223;
+            gDPFillRectangle(gfxP++, ulx, uly, lrx, lry);
+            gDPPipeSync(gfxP++);
+        }
+    };
+
+    *gfx = gfxP;
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/game/GameState_DrawInputDisplay.s")
+#endif
+
+#ifdef NON_MATCHING
+// very close, dispRefs are off by 0x4 bytes
+void GameState_Draw(GameState* gameState, GraphicsContext* gfxCtx) {
+    Gfx* newDList;
+    Gfx* polyOpaP;
+    Gfx* dispRefs[4];
+    GfxPrint printChars[3];
+
+    Graph_OpenDisps(dispRefs, gfxCtx, "../game.c", 746);
+    do {
+        newDList = Graph_GfxPlusOne(polyOpaP = gfxCtx->polyOpa.p);
+        gSPDisplayList(gfxCtx->overlay.p++, newDList);
+    } while (0);
+
+    if (R_ENABLE_FB_FILTER == 1) {
+        GameState_SetFBFilter(&newDList);
+    }
+
+    sLastButtonPressed = gameState->input[0].press.in.button | gameState->input[0].cur.in.button;
+    if (R_DISABLE_INPUT_DISPLAY == 0) {
+        GameState_DrawInputDisplay(sLastButtonPressed, &newDList);
+    }
+
+    if (R_ENABLE_AUDIO_DBG & 1) {
+        GfxPrint_Ctor(printChars);
+        GfxPrint_Open(printChars, newDList);
+        func_800EEA50(printChars);
+        newDList = GfxPrint_Close(printChars);
+        GfxPrint_Dtor(printChars);
+    }
+
+    if (R_ENABLE_ARENA_DBG < 0) {
+        DebugArena_Display();
+        SystemArena_Display();
+        //% 08x bytes left until the death of Hyrule (game_alloc)
+        osSyncPrintf("ハイラル滅亡まであと %08x バイト(game_alloc)\n", THA_GetSize(&gameState->tha));
+        R_ENABLE_ARENA_DBG = 0;
+    }
+
+    do {
+        gSPEndDisplayList(newDList++);
+        Graph_BranchDlist(polyOpaP, newDList);
+        gfxCtx->polyOpa.p = newDList;
+    } while (0);
```

Comment:
Can any of these do while (0) be replaced by if (1)s / if (0)s? If possible empty, but either way it would already be better than a do while (0).